### ISSUE TITLE
Fresh Grails 3.3.2 grails run-app does not start

### ISSUE THREAD

**Issue by erichelgeson** [grails/grails-core#10877]
### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [ ] Example that reproduces the problem uploaded to Github
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

```bash
$ sdk use grails 3.3.2
Using grails version 3.3.2 in this shell.
$ grails create-app foobar
| Application created at /Users/erichelgeson/source/foobar
$ cd foobar
$ grails run-app
| Resolving Dependencies. Please wait...
Cannot resolve reloading agent JAR: Could not resolve all dependencies for configuration ':agent'.

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/erichelgeson/source/foobar/build.gradle' line: 22

* What went wrong:
A problem occurred evaluating root project 'foobar'.
> Failed to apply plugin [id 'com.energizedwork.webdriver-binaries']
   > No signature of method: java.lang.String.uncapitalize() is applicable for argument types: () values: []
     Possible solutions: capitalize()

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

CONFIGURE FAILED

Total time: 5.535 secs
| Error Error initializing classpath: No signature of method: java.lang.String.uncapitalize() is applicable for argument types: () values: []
Possible solutions: capitalize() (Use --stacktrace to see the full trace)
```

### Expected Behaviour

App should run

### Actual Behaviour

App exits with exception

### Work around

Note `bootRun` does work.
Remove new gradle plugin causing issues `com.energizedwork.webdriver-binaries`

### Environment Information

- **Operating System**: `MacOS 10.12.6`
- **Grails Version:** 3.3.2
- **JDK Version:** 8u111
- **Container Version (If Applicable):** n/a

### Example Application

n/a

**Comment by sdelamo** [comment]
The `webdriver-binaries` Gradle plugin uses uncapitalize. Uncapitalize is included in Groovy since 2.4.8. Grails 3.3.2 uses Groovy 2.4.13 and Gradle 3.5. Somehow a lower version of Groovy is being picked up.

**Comment by graemerocher** [comment]
Check your current installed version of Gradle via SDKman. It may be the version you have is incompatible.

**Comment by erichelgeson** [comment]
```
$ groovy -v
Groovy Version: 2.4.13 JVM: 1.8.0_111 Vendor: Oracle Corporation OS: Mac OS X
$ which groovy
/Users/erichelgeson/.sdkman/candidates/groovy/current/bin/groovy

$ grails -v
| Grails Version: 3.3.2
| Groovy Version: 2.4.13
| JVM Version: 1.8.0_111
$ which grails
/Users/erichelgeson/.sdkman/candidates/grails/3.3.2/bin/grails
```

I'd hope the installed version of groovy wouldn't have an effect on the grails commands. 

I also uninstalled all versions of groovy and the error still happens. 

Same error happens with `./grailsw` as well.

**Comment by jeffbrown** [comment]
@erichelgeson I think Graeme was coming on your current install version of Gradle.

**Comment by jeffbrown** [comment]
I have tried to reproduce the problem with various versions of dependencies configured in my environment and I haven't been able to reproduce it yet.

**Comment by erichelgeson** [comment]
@jeffbrown thanks, misread that.

```
$ gradle -v
------------------------------------------------------------
Gradle 3.4.1
------------------------------------------------------------
Groovy:       2.4.7 <-- old
Ant:          Apache Ant(TM) version 1.9.6 compiled on June 29 2015
JVM:          1.8.0_111 (Oracle Corporation 25.111-b14)
OS:           Mac OS X 10.12.6 x86_64
```

I `$ sdk uninstall gradle 3.4.1` then it does start up when no gradle in my path. 

Though I find it unexpected it uses the system installed gradle vs whatever grailsw or grails command requires.

**Comment by sdelamo** [comment]
Gradle 3.4.1 shipped with Groovy 2.4.7. Where `String.uncapitalize() `did not existed. So I am going to close this issue. Thanks for reporting @erichelgeson I will keep an eye on slack and Github Issues to see if other users are experiencing the issue and point them to upgrade their local gradle version to 3.5.

**Comment by erichelgeson** [comment]
This behavior is unexpected to me:
`grails` with no gradle set - unsure which gradle binary it uses.
`grails` || `./gailsw` with gradle on path - uses the one on the path, which could be much older or newer than shipped with Grails.

I'd expect using the `grails` commands I'd expect Grails to use the version it shipped with.

If I wanted to use a specific version of gradle I'd use `gradle` or `gradlew`.

Having groovy or groovy home set on my path does not affect the Grails commands.

**Comment by graemerocher** [comment]
The behaviour you are seeing has been there a long time.  Grails aligns the version used with your current GVM version if it is not set. See https://github.com/grails/grails-core/blob/master/grails-shell/src/main/groovy/org/grails/cli/gradle/GradleUtil.groovy#L53

Otherwise it uses the Gradle version that Grails is built against. You can set a fixed version by setting `GRAILS_GRADLE_HOME`

**Comment by erichelgeson** [comment]
Does sdkman need a special case for this? `sdk use gradle 3.4.1` will set `GRADLE_HOME`. If Grails used this it would then work for any OS. (I don't think that would be a good thing, for the same reasons I don't think sdkman current is a good thing.)

As a novice user I'd want a fresh grails app to just work. (use built against gradle)

As an advanced user I'd want to be able to able to override it. (GRAILS_GRADLE_HOME)

(This is really not a huge deal, I'm just advocating for a better default. Paper cuts for newcomers to Grails)

**Comment by erichelgeson** [comment]
After talking to @jameskleeh at G3Summit this should probably be re-opened as the behavior is odd and provides indication of whats happening to the user. I actually saw 2 people at the conference have this issue, and have seen a few people in slack have it as well.

**Comment by sdelamo** [comment]
Hello @erichelgeson 

For Grails 3.3.3, there will be two features: 

`geb` - JDK7 compatible (geb 1.1.1, webdriver 2) no webdriver-binaries plugin

and `geb2` - JDK8 compatible (geb 2, webdriver 3.6), webdriver-binaries plugin

default feature will be `geb` to reduce errors such as the one you described here.

**Comment by ppazos** [comment]
I experienced this issue having Gradle 5 installed via SDKMAN! and my Grails 3.2.9 project was using Gradle 3.4.1 (version in gradle.properties). Installed Gradle 3.4.1 and selected as current on SDKMAN! and the run-app worked again.

As @erichelgeson mentioned above, it would be good if Grails has it's own Gradle independent of what is configured on SDKMAN! is a pain to change the Grails version and the Gradle version for every project (I have many projects on different Grails versions).

**Comment by lalitvchau** [comment]
This error to due to maven was installed in you system..

follow the below steps 
1. delete the c:\Users\.m2 or related directory in windows (Not sure about Unix or Mac but you can find the local cache of maven. Delete .m2 directory)
 2. delete maven related environment variable. 
3. delete ./gradlew 
4. delete environment variable related to groovy also

try to above step.. it will work fine

**Comment by jamesdh** [comment]
@erichelgeson I believe your concerns with this were ultimately resolved by #10908
